src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -250,45 +251,41 @@ public void run() {
     driver.run();
   }
 
-  private boolean launchNode(SchedulerDriver driver, Offer offer,
-        String nodeName, List<String> taskTypes, String executorName) {
+  private void launchNode(SchedulerDriver driver, Offer offer,
+        String nodeName, Map<String,String> taskInfo, String executorName) {
     //nodeName is the type of executor to launch
     //executorName is to distinguish different types of nodes
-    //taskType is the type of task in mesos to launch on the node
-    //taskName is a name chosen to identify the task in mesos and mesos-dns (if used)
+    //taskInfo is taskType mapped to taskName
     log.info(String.format("Launching node of type %s with tasks %s", nodeName,
-        taskTypes.toString()));
+        taskInfo));
     String taskIdName = String.format("%s.%s.%d", nodeName, executorName,
         System.currentTimeMillis());
     List<Resource> resources = getExecutorResources();
     ExecutorInfo executorInfo = createExecutor(taskIdName, nodeName, executorName, resources);
     List<TaskInfo> tasks = new ArrayList<>();
-    for (String taskType : taskTypes) {
-      List<Resource> taskResources = getTaskResources(taskType);
-      String taskName = getNextTaskType(taskType);
-      if (taskName.isEmpty()) return false;
+    for (Map.Entry<String,String> taskItem : taskInfo.entrySet()) {
```

Comment:
I disagree. entrySet is of type Set<Map.Entry<String,String>>. That type should be implicit in the code, not explicitly declared.